### ISSUE TITLE
kubectl exec fails on  v1.6.1 to any pod

### ISSUE THREAD

**Issue by csarora** [kubernetes/kubernetes#44662]
OS: Centos 7 
Vsphere

Kubeadm init --kubernetes-version v1.6.1 successful 
sudo cp /etc/kubernetes/admin.conf $HOME/
sudo chown $(id -u):$(id -g) $HOME/admin.conf
export KUBECONFIG=$HOME/admin.conf

But kubectl exec errors out due to below error, please help

```
[root@dotricorder03-master-01 templates]# export KUBECONFIG=$HOME/admin.conf
[root@dotricorder03-master-01 templates]# **kubectl exec -it zookeeper-0 bash**
error: error sending request: Post https://172.22.16.69:6443/api/v1/namespaces/default/pods/zookeeper-0/exec?command=bash&container=zookeeper&container=zookeeper&stdin=true&stdout=true&tty=true: EOF
```

**Comment by adohe** [comment]
@csarora could you please use `kubectl exec --v=9` to provide more info?

**Comment by csarora** [comment]
`[root@dotricorder04-master-01 ~]# kubectl exec --v=9 -it zookeeper-0 /bin/bash
I0419 10:34:12.159282   11953 loader.go:354] Config loaded from file /etc/kubernetes/admin.conf
I0419 10:34:12.160868   11953 cached_discovery.go:118] returning cached discovery info from /root/.kube/cache/discovery/172.22.16.69_6443/servergroups.json
I0419 10:34:12.161336   11953 round_trippers.go:398] curl -k -v -XGET  -H "Accept: application/json, */*" -H "User-Agent: kubectl/v1.6.1 (linux/amd64) kubernetes/b0b7a32" https://172.22.16.69:6443/api/v1/namespaces/default/pods/zookeeper-0
I0419 10:34:12.203409   11953 round_trippers.go:417] GET https://172.22.16.69:6443/api/v1/namespaces/default/pods/zookeeper-0 200 OK in 42 milliseconds
I0419 10:34:12.203512   11953 round_trippers.go:423] Response Headers:
I0419 10:34:12.203524   11953 round_trippers.go:426]     Content-Type: application/json
I0419 10:34:12.203536   11953 round_trippers.go:426]     Content-Length: 3146
I0419 10:34:12.203541   11953 round_trippers.go:426]     Date: Wed, 19 Apr 2017 10:34:12 GMT
I0419 10:34:12.203659   11953 request.go:991] Response Body: {"kind":"Pod","apiVersion":"v1","metadata":{"name":"zookeeper-0","generateName":"zookeeper-","namespace":"default","selfLink":"/api/v1/namespaces/default/pods/zookeeper-0","uid":"8900ccb9-24eb-11e7-a738-005056916c4e","resourceVersion":"790","creationTimestamp":"2017-04-19T10:32:49Z","labels":{"app":"zookeeper"},"annotations":{"kubernetes.io/created-by":"{\"kind\":\"SerializedReference\",\"apiVersion\":\"v1\",\"reference\":{\"kind\":\"StatefulSet\",\"namespace\":\"default\",\"name\":\"zookeeper\",\"uid\":\"88ffda71-24eb-11e7-a738-005056916c4e\",\"apiVersion\":\"apps\",\"resourceVersion\":\"721\"}}\n","pod.beta.kubernetes.io/hostname":"zookeeper-0","pod.beta.kubernetes.io/subdomain":"zookeeper"},"ownerReferences":[{"apiVersion":"apps/v1beta1","kind":"StatefulSet","name":"zookeeper","uid":"88ffda71-24eb-11e7-a738-005056916c4e","co
ntroller":true,"blockOwnerDeletion":true}]},"spec":{"volumes":[{"name":"datadir","emptyDir":{}},{"name":"default-token-w7k6k","secret":{"secretName":"default-token-w7k6k","defaultMode":420}}],"containers":[{"name":"zookeeper","image":"devhub-docker.cisco.com/vnf-docker-dev/zookeeper:3.4.9.9.b3162086","ports":[{"name":"client","containerPort":2181,"protocol":"TCP"},{"name":"peer","containerPort":2888,"protocol":"TCP"},{"name":"leader-election","containerPort":3888,"protocol":"TCP"}],"env":[{"name":"ZOO_SERVERS","value":"server.1=zookeeper-0.zookeeper:2888:3888:participant"}],"resources":{},"volumeMounts":[{"name":"datadir","mountPath":"/data"},{"name":"default-token-w7k6k","readOnly":true,"mountPath":"/var/run/secrets/kubernetes.io/serviceaccount"}],"terminationMessagePath":"/dev/termination-log","terminationMessagePolicy":"File","imagePullPolicy":"Always"}],"restartPolicy":"Always","term
inationGracePeriodSeconds":10,"dnsPolicy":"ClusterFirst","serviceAccountName":"default","serviceAccount":"default","nodeName":"dotricorder04-master-01","securityContext":{},"imagePullSecrets":[{"name":"tricregkey"}],"schedulerName":"default-scheduler","tolerations":[{"key":"node.alpha.kubernetes.io/notReady","operator":"Exists","effect":"NoExecute","tolerationSeconds":300},{"key":"node.alpha.kubernetes.io/unreachable","operator":"Exists","effect":"NoExecute","tolerationSeconds":300}]},"status":{"phase":"Running","conditions":[{"type":"Initialized","status":"True","lastProbeTime":null,"lastTransitionTime":"2017-04-19T10:32:49Z"},{"type":"Ready","status":"True","lastProbeTime":null,"lastTransitionTime":"2017-04-19T10:33:23Z"},{"type":"PodScheduled","status":"True","lastProbeTime":null,"lastTransitionTime":"2017-04-19T10:32:49Z"}],"hostIP":"172.22.16.69","podIP":"10.32.0.4","startTime":"201
7-04-19T10:32:49Z","containerStatuses":[{"name":"zookeeper","state":{"running":{"startedAt":"2017-04-19T10:33:23Z"}},"lastState":{},"ready":true,"restartCount":0,"image":"devhub-docker.cisco.com/vnf-docker-dev/zookeeper:3.4.9.9.b3162086","imageID":"docker-pullable://devhub-docker.cisco.com/vnf-docker-dev/zookeeper@sha256:2368500d99a44cc3bcd4f6a1a52b5af848e7bb746af9150b5c966cde51d41e12","containerID":"docker://df54fc7d0e02fa1f2fe9893d29aea528f6055ea326d4354debbd9732652d4e09"}],"qosClass":"BestEffort"}}
I0419 10:34:12.205921   11953 round_trippers.go:398] curl -k -v -XPOST  -H "X-Stream-Protocol-Version: v4.channel.k8s.io" -H "X-Stream-Protocol-Version: v3.channel.k8s.io" -H "X-Stream-Protocol-Version: v2.channel.k8s.io" -H "X-Stream-Protocol-Version: channel.k8s.io" https://172.22.16.69:6443/api/v1/namespaces/default/pods/zookeeper-0/exec?command=%2Fbin%2Fbash&container=zookeeper&container=zookeeper&stdin=true&stdout=true&tty=true
I0419 10:34:12.205971   11953 round_trippers.go:398] curl -k -v -XPOST  -H "X-Stream-Protocol-Version: v4.channel.k8s.io" -H "X-Stream-Protocol-Version: v3.channel.k8s.io" -H "X-Stream-Protocol-Version: v2.channel.k8s.io" -H "X-Stream-Protocol-Version: channel.k8s.io" -H "User-Agent: kubectl/v1.6.1 (linux/amd64) kubernetes/b0b7a32" https://172.22.16.69:6443/api/v1/namespaces/default/pods/zookeeper-0/exec?command=%2Fbin%2Fbash&container=zookeeper&container=zookeeper&stdin=true&stdout=true&tty=true
I0419 10:34:12.209884   11953 round_trippers.go:417] POST https://172.22.16.69:6443/api/v1/namespaces/default/pods/zookeeper-0/exec?command=%2Fbin%2Fbash&container=zookeeper&container=zookeeper&stdin=true&stdout=true&tty=true  in 3 milliseconds
I0419 10:34:12.209898   11953 round_trippers.go:423] Response Headers:
I0419 10:34:12.209904   11953 round_trippers.go:417] POST https://172.22.16.69:6443/api/v1/namespaces/default/pods/zookeeper-0/exec?command=%2Fbin%2Fbash&container=zookeeper&container=zookeeper&stdin=true&stdout=true&tty=true  in 3 milliseconds
I0419 10:34:12.209908   11953 round_trippers.go:423] Response Headers:
F0419 10:34:12.210123   11953 helpers.go:119] error: error sending request: Post https://172.22.16.69:6443/api/v1/namespaces/default/pods/zookeeper-0/exec?command=%2Fbin%2Fbash&container=zookeeper&container=zookeeper&stdin=true&stdout=true&tty=true: EOF
[root@dotricorder04-master-01 ~]#
`

**Comment by dchen1107** [comment]
cc/ @timstclair

**Comment by ncdc** [comment]
@csarora are there any errors in either the apiserver or kubelet logs around the time you issued the exec?

**Comment by csarora** [comment]
[root@dotricorder03-master-01 ~]# kubelet logs
I0419 17:01:12.909410   15460 feature_gate.go:144] feature gates: map[]
W0419 17:01:12.910191   15460 server.go:469] No API client: no api servers specified
I0419 17:01:12.910232   15460 docker.go:364] Connecting to docker on unix:///var/run/docker.sock
I0419 17:01:12.910252   15460 docker.go:384] Start docker client with request timeout=2m0s
I0419 17:01:12.952336   15460 manager.go:143] cAdvisor running in container: "/user.slice"
W0419 17:01:13.024517   15460 manager.go:151] unable to connect to Rkt api service: rkt: cannot tcp Dial rkt api service: dial tcp 127.0.0.1:15441: getsockopt: connection refused
I0419 17:01:13.094716   15460 fs.go:117] Filesystem partitions: map[/dev/mapper/cl-root:{mountpoint:/ major:253 minor:0 fsType:xfs blockSize:0} /dev/mapper/cl-home:{mountpoint:/home major:253 minor:2 fsType:xfs blockSize:0} /dev/sda1:{mountpoint:/boot major:8 minor:1 fsType:xfs blockSize:0}]
I0419 17:01:13.104682   15460 manager.go:198] Machine: {NumCores:16 CpuFrequency:2999692 MemoryCapacity:33567109120 MachineID:4f79f76df37244cfa49a732da9b499b0 SystemUUID:F99F1142-461D-2684-1785-AB7EC9791354 BootID:56a2b4cb-bf20-4464-9acb-de846ceec959 Filesystems:[{Device:/dev/sda1 Capacity:1063256064 Type:vfs Inodes:524288 HasInodes:true} {Device:/dev/mapper/cl-root Capacity:51200024576 Type:vfs Inodes:25012224 HasInodes:true} {Device:/dev/mapper/cl-home Capacity:24994230272 Type:vfs Inodes:12210176 HasInodes:true}] DiskMap:map[253:11:{Name:dm-11 Major:253 Minor:11 Size:10737418240 Scheduler:none} 253:12:{Name:dm-12 Major:253 Minor:12 Size:10737418240 Scheduler:none} 253:14:{Name:dm-14 Major:253 Minor:14 Size:10737418240 Scheduler:none} 253:19:{Name:dm-19 Major:253 Minor:19 Size:10737418240 Scheduler:none} 253:1:{Name:dm-1 Major:253 Minor:1 Size:8589934592 Scheduler:none} 253:29:{Name:dm
-29 Major:253 Minor:29 Size:10737418240 Scheduler:none} 253:8:{Name:dm-8 Major:253 Minor:8 Size:10737418240 Scheduler:none} 253:9:{Name:dm-9 Major:253 Minor:9 Size:10737418240 Scheduler:none} 253:0:{Name:dm-0 Major:253 Minor:0 Size:51225034752 Scheduler:none} 253:2:{Name:dm-2 Major:253 Minor:2 Size:25006440448 Scheduler:none} 253:20:{Name:dm-20 Major:253 Minor:20 Size:10737418240 Scheduler:none} 253:30:{Name:dm-30 Major:253 Minor:30 Size:10737418240 Scheduler:none} 253:5:{Name:dm-5 Major:253 Minor:5 Size:10737418240 Scheduler:none} 253:10:{Name:dm-10 Major:253 Minor:10 Size:10737418240 Scheduler:none} 253:18:{Name:dm-18 Major:253 Minor:18 Size:10737418240 Scheduler:none} 253:26:{Name:dm-26 Major:253 Minor:26 Size:10737418240 Scheduler:none} 253:32:{Name:dm-32 Major:253 Minor:32 Size:10737418240 Scheduler:none} 253:6:{Name:dm-6 Major:253 Minor:6 Size:10737418240 Scheduler:none} 253:31:{Na
me:dm-31 Major:253 Minor:31 Size:10737418240 Scheduler:none} 8:0:{Name:sda Major:8 Minor:0 Size:85899345920 Scheduler:deadline} 253:23:{Name:dm-23 Major:253 Minor:23 Size:10737418240 Scheduler:none} 253:24:{Name:dm-24 Major:253 Minor:24 Size:10737418240 Scheduler:none} 253:25:{Name:dm-25 Major:253 Minor:25 Size:10737418240 Scheduler:none} 253:27:{Name:dm-27 Major:253 Minor:27 Size:10737418240 Scheduler:none} 253:4:{Name:dm-4 Major:253 Minor:4 Size:10737418240 Scheduler:none} 253:7:{Name:dm-7 Major:253 Minor:7 Size:10737418240 Scheduler:none} 253:16:{Name:dm-16 Major:253 Minor:16 Size:10737418240 Scheduler:none} 253:17:{Name:dm-17 Major:253 Minor:17 Size:10737418240 Scheduler:none} 253:21:{Name:dm-21 Major:253 Minor:21 Size:10737418240 Scheduler:none} 253:33:{Name:dm-33 Major:253 Minor:33 Size:10737418240 Scheduler:none} 253:13:{Name:dm-13 Major:253 Minor:13 Size:10737418240 Scheduler:non
e} 253:15:{Name:dm-15 Major:253 Minor:15 Size:10737418240 Scheduler:none} 253:22:{Name:dm-22 Major:253 Minor:22 Size:10737418240 Scheduler:none} 253:28:{Name:dm-28 Major:253 Minor:28 Size:10737418240 Scheduler:none} 253:3:{Name:dm-3 Major:253 Minor:3 Size:107374182400 Scheduler:none}] NetworkDevices:[{Name:datapath MacAddress:86:03:bc:35:d9:a6 Speed:0 Mtu:1376} {Name:dummy0 MacAddress:c2:0a:6d:8e:a8:42 Speed:0 Mtu:1500} {Name:ens192 MacAddress:00:50:56:91:97:05 Speed:10000 Mtu:1500} {Name:vxlan-6784 MacAddress:d6:3f:b7:e1:40:16 Speed:0 Mtu:65470} {Name:weave MacAddress:e2:fe:86:af:cb:d5 Speed:0 Mtu:1376}] Topology:[{Id:0 Memory:17179336704 Cores:[{Id:0 Threads:[0] Caches:[{Size:32768 Type:Data Level:1} {Size:32768 Type:Instruction Level:1} {Size:262144 Type:Unified Level:2}]}] Caches:[{Size:26214400 Type:Unified Level:3}]} {Id:2 Memory:0 Cores:[{Id:0 Threads:[1] Caches:[{Size:32768 Type:
Data Level:1} {Size:32768 Type:Instruction Level:1} {Size:262144 Type:Unified Level:2}]}] Caches:[{Size:26214400 Type:Unified Level:3}]} {Id:4 Memory:0 Cores:[{Id:0 Threads:[2] Caches:[{Size:32768 Type:Data Level:1} {Size:32768 Type:Instruction Level:1} {Size:262144 Type:Unified Level:2}]}] Caches:[{Size:26214400 Type:Unified Level:3}]} {Id:6 Memory:0 Cores:[{Id:0 Threads:[3] Caches:[{Size:32768 Type:Data Level:1} {Size:32768 Type:Instruction Level:1} {Size:262144 Type:Unified Level:2}]}] Caches:[{Size:26214400 Type:Unified Level:3}]} {Id:8 Memory:0 Cores:[{Id:0 Threads:[4] Caches:[{Size:32768 Type:Data Level:1} {Size:32768 Type:Instruction Level:1} {Size:262144 Type:Unified Level:2}]}] Caches:[{Size:26214400 Type:Unified Level:3}]} {Id:10 Memory:0 Cores:[{Id:0 Threads:[5] Caches:[{Size:32768 Type:Data Level:1} {Size:32768 Type:Instruction Level:1} {Size:262144 Type:Unified Level:2}]}] C
aches:[{Size:26214400 Type:Unified Level:3}]} {Id:12 Memory:0 Cores:[{Id:0 Threads:[6] Caches:[{Size:32768 Type:Data Level:1} {Size:32768 Type:Instruction Level:1} {Size:262144 Type:Unified Level:2}]}] Caches:[{Size:26214400 Type:Unified Level:3}]} {Id:14 Memory:0 Cores:[{Id:0 Threads:[7] Caches:[{Size:32768 Type:Data Level:1} {Size:32768 Type:Instruction Level:1} {Size:262144 Type:Unified Level:2}]}] Caches:[{Size:26214400 Type:Unified Level:3}]} {Id:16 Memory:0 Cores:[{Id:0 Threads:[8] Caches:[{Size:32768 Type:Data Level:1} {Size:32768 Type:Instruction Level:1} {Size:262144 Type:Unified Level:2}]}] Caches:[{Size:26214400 Type:Unified Level:3}]} {Id:18 Memory:0 Cores:[{Id:0 Threads:[9] Caches:[{Size:32768 Type:Data Level:1} {Size:32768 Type:Instruction Level:1} {Size:262144 Type:Unified Level:2}]}] Caches:[{Size:26214400 Type:Unified Level:3}]} {Id:20 Memory:0 Cores:[{Id:0 Threads:[10]
Caches:[{Size:32768 Type:Data Level:1} {Size:32768 Type:Instruction Level:1} {Size:262144 Type:Unified Level:2}]}] Caches:[{Size:26214400 Type:Unified Level:3}]} {Id:22 Memory:0 Cores:[{Id:0 Threads:[11] Caches:[{Size:32768 Type:Data Level:1} {Size:32768 Type:Instruction Level:1} {Size:262144 Type:Unified Level:2}]}] Caches:[{Size:26214400 Type:Unified Level:3}]} {Id:24 Memory:0 Cores:[{Id:0 Threads:[12] Caches:[{Size:32768 Type:Data Level:1} {Size:32768 Type:Instruction Level:1} {Size:262144 Type:Unified Level:2}]}] Caches:[{Size:26214400 Type:Unified Level:3}]} {Id:26 Memory:0 Cores:[{Id:0 Threads:[13] Caches:[{Size:32768 Type:Data Level:1} {Size:32768 Type:Instruction Level:1} {Size:262144 Type:Unified Level:2}]}] Caches:[{Size:26214400 Type:Unified Level:3}]} {Id:28 Memory:0 Cores:[{Id:0 Threads:[14] Caches:[{Size:32768 Type:Data Level:1} {Size:32768 Type:Instruction Level:1} {Size:2
62144 Type:Unified Level:2}]}] Caches:[{Size:26214400 Type:Unified Level:3}]} {Id:30 Memory:0 Cores:[{Id:0 Threads:[15] Caches:[{Size:32768 Type:Data Level:1} {Size:32768 Type:Instruction Level:1} {Size:262144 Type:Unified Level:2}]}] Caches:[{Size:26214400 Type:Unified Level:3}]}] CloudProvider:Unknown InstanceType:Unknown InstanceID:None}
I0419 17:01:13.132614   15460 manager.go:204] Version: {KernelVersion:3.10.0-514.2.2.el7.x86_64 ContainerOsVersion:CentOS Linux 7 (Core) DockerVersion:1.12.5 CadvisorVersion: CadvisorRevision:}
W0419 17:01:13.133359   15460 server.go:350] No api server defined - no events will be sent to API server.
I0419 17:01:13.133374   15460 server.go:509] --cgroups-per-qos enabled, but --cgroup-root was not specified.  defaulting to /
I0419 17:01:13.133736   15460 cadvisor_linux.go:152] Failed to register cAdvisor on port 4194, retrying. Error: listen tcp :4194: bind: address already in use
W0419 17:01:13.135100   15460 container_manager_linux.go:218] Running with swap on is not supported, please disable swap! This will be a fatal error by default starting in K8s v1.6! In the meantime, you can opt-in to making this a fatal error by enabling --experimental-fail-swap-on.
I0419 17:01:13.135181   15460 container_manager_linux.go:245] container manager verified user specified cgroup-root exists: /
I0419 17:01:13.135193   15460 container_manager_linux.go:250] Creating Container Manager object based on Node Config: {RuntimeCgroupsName: SystemCgroupsName: KubeletCgroupsName: ContainerRuntime:docker CgroupsPerQOS:true CgroupRoot:/ CgroupDriver:cgroupfs ProtectKernelDefaults:false EnableCRI:true NodeAllocatableConfig:{KubeReservedCgroupName: SystemReservedCgroupName: EnforceNodeAllocatable:map[pods:{}] KubeReserved:map[] SystemReserved:map[] HardEvictionThresholds:[{Signal:memory.available Operator:LessThan Value:{Quantity:100Mi Percentage:0} GracePeriod:0s MinReclaim:<nil>}]} ExperimentalQOSReserved:map[]}
W0419 17:01:13.137640   15460 kubelet_network.go:70] Hairpin mode set to "promiscuous-bridge" but kubenet is not enabled, falling back to "hairpin-veth"
I0419 17:01:13.137688   15460 kubelet.go:494] Hairpin mode set to "hairpin-veth"
I0419 17:01:13.147936   15460 docker_service.go:187] Docker cri networking managed by kubernetes.io/no-op
error: failed to run Kubelet: failed to create kubelet: misconfiguration: kubelet cgroup driver: "cgroupfs" is different from docker cgroup driver: "systemd"
[root@dotricorder03-master-01 ~]#

**Comment by csarora** [comment]
Actually everything is getting created successfully only exec is giving this error, is it something to do with port or reaching api?? firewall is disabled so i am not sure whats causing this exec to fail

> [root@dotricorder03-master-01 ~]# kubectl get po --all-namespaces
NAMESPACE     NAME                                              READY     STATUS    RESTARTS   AGE
default       kafka-0                                           1/1       Running   0          18m
default       kafkaclient                                       1/1       Running   0          18m
default       redis-master                                      2/2       Running   0          7m
default       redis-sentinel-n4v92                              1/1       Running   0          7m
default       zookeeper-0                                       1/1       Running   0          18m
kube-system   etcd-dotricorder03-master-01                      1/1       Running   0          19m
kube-system   kube-apiserver-dotricorder03-master-01            1/1       Running   2          19m
kube-system   kube-controller-manager-dotricorder03-master-01   1/1       Running   0          20m
kube-system   kube-dns-3913472980-wvxnr                         3/3       Running   0          20m
kube-system   kube-proxy-r8t0l                                  1/1       Running   0          20m
kube-system   kube-scheduler-dotricorder03-master-01            1/1       Running   0          20m
kube-system   kubernetes-dashboard-2457468166-j11qr             1/1       Running   0          20m
kube-system   weave-net-p4674                                   2/2       Running   1          20m
[root@dotricorder03-master-01 ~]#

>

**Comment by ncdc** [comment]
The kubelet logs you pasted were incorrect - you need to get the actual logs, not the output from running `kubelet logs`. Look in /var/log or the journal.

**Comment by csarora** [comment]
API Server logs attached 

2017-04-19T16:42:48.479520498Z E0419 16:42:48.479340       1 reflector.go:201] k8s.io/kubernetes/pkg/client/informers/informers_generated/internalversion/factory.go:70: Failed to list *rbac.RoleBinding: **Get https://localhost:6443/apis/rbac.authorization.k8s.io/v1beta1/rolebindings?resourceVersion=0: dial tcp 127.0.0.1:6443: getsockopt: connection refused**
2017-04-19T16:42:48.504677147Z [restful] 2017/04/19 16:42:48 log.go:30: [restful/swagger] listing is available at https://172.22.16.69:6443/swaggerapi/
2017-04-19T16:42:48.504753876Z [restful] 2017/04/19 16:42:48 log.go:30: [restful/swagger] https://172.22.16.69:6443/swaggerui/ is mapped to folder /swagger-ui/
2017-04-19T16:42:48.562595006Z I0419 16:42:48.562374       1 serve.go:79] **Serving securely on 0.0.0.0:6443**
2017-04-19T16:42:49.501006836Z I0419 16:42:49.500842       1 trace.go:61] Trace "Create /api/v1/namespaces/kube-system/configmaps" (started 2017-04-19 16:42:48.596676092 +0000 UTC):
2017-04-19T16:42:49.50103654Z [71.127µs] [71.127µs] About to convert to expected version
2017-04-19T16:42:49.501041319Z [153.145µs] [82.018µs] Conversion done
2017-04-19T16:42:49.501044385Z [903.158449ms] [903.005304ms] About to store object in database
2017-04-19T16:42:49.501047239Z [904.050905ms] [892.456µs] Object stored in database
2017-04-19T16:42:49.501050032Z [904.065018ms] [14.113µs] Self-link added
2017-04-19T16:42:49.501052708Z "Create /api/v1/namespaces/kube-system/configmaps" [904.120531ms] [55.513µs] END

[logs-apiserver.txt](https://github.com/kubernetes/kubernetes/files/936672/logs-apiserver.txt)

**Comment by ncdc** [comment]
Either your apiserver loglevel is very low or you didn't capture everything - there's not really anything in there. How about the kubelet?

**Comment by csarora** [comment]
Unable to find any logs for kubelet under /var/log/.....is there any other place to look for?

**Comment by ncdc** [comment]
I am not familiar with how kubeadm sets things up. How is the kubelet running? Does `sudo systemctl list-units | grep kubelet` yield anything? If so, try `sudo journalctl -u kubelet`.

**Comment by csarora** [comment]
root@dotricorder03-master-01 containers]# sudo journalctl -u kubelet
-- Logs begin at Wed 2017-04-19 16:34:49 UTC, end at Wed 2017-04-19 17:30:19 UTC. --
...
Apr 19 16:39:01 dotricorder03-master-01 kubelet[4556]: I0419 16:39:01.293619    4556 feature_gate.go:144] feature gates: map[]
Apr 19 16:39:01 dotricorder03-master-01 kubelet[4556]: error: failed to run Kubelet: invalid kubeconfig: stat /etc/kubernetes/kubelet.conf: no such file or directory
Apr 19 16:39:01 dotricorder03-master-01 systemd[1]: kubelet.service: main process exited, code=exited, status=1/FAILURE
Apr 19 16:39:01 dotricorder03-master-01 systemd[1]: Unit kubelet.service entered failed state.
Apr 19 16:39:01 dotricorder03-master-01 systemd[1]: kubelet.service failed.
Apr 19 16:39:11 dotricorder03-master-01 systemd[1]: kubelet.service holdoff time over, scheduling restart.
Apr 19 16:39:11 dotricorder03-master-01 systemd[1]: Started kubelet: The Kubernetes Node Agent.
Apr 19 16:39:11 dotricorder03-master-01 systemd[1]: Starting kubelet: The Kubernetes Node Agent...
Apr 19 16:39:11 dotricorder03-master-01 kubelet[4604]: I0419 16:39:11.607232    4604 feature_gate.go:144] feature gates: map[]
Apr 19 16:39:11 dotricorder03-master-01 kubelet[4604]: I0419 16:39:11.834209    4604 docker.go:364] Connecting to docker on unix:///var/run/docker.sock
Apr 19 16:39:11 dotricorder03-master-01 kubelet[4604]: I0419 16:39:11.834243    4604 docker.go:384] Start docker client with request timeout=2m0s
Apr 19 16:39:11 dotricorder03-master-01 kubelet[4604]: W0419 16:39:11.878123    4604 cni.go:157] Unable to update cni config: No networks found in /etc/cni/net.d
Apr 19 16:39:11 dotricorder03-master-01 kubelet[4604]: I0419 16:39:11.888622    4604 manager.go:143] cAdvisor running in container: "/"
Apr 19 16:39:11 dotricorder03-master-01 kubelet[4604]: W0419 16:39:11.947297    4604 manager.go:151] unable to connect to Rkt api service: rkt: cannot tcp Dial rkt api service: dial tcp 127.0.0.1:15441: Apr 19 16:39:12 dotricorder03-master-01 kubelet[4604]: I041
 Level:1} {Size:262144 Type:Unified Level:2}]}] Caches:[{Size:26214400 Type:Unified Level:3}]}] CloudProvider:UnknoApr 19 16:39:12 dotricorder03-master-01 kubelet[4604]: I0419 16:39:12.033958    4604 manager.go:204] Version: {KernelVersion:3.10.0-514.2.2.el7.x86
Apr 19 16:39:12 dotricorder03-master-01 kubelet[4604]: W0419 16:39:12.036409    4604 container_manager_linux.go:218] Running with swap on is not supported, please disable swap! This will be a fatal errorApr 19 16:39:12 dotricorder03-master-01 kubelet[4604]: I041
Apr 19 16:39:12 dotricorder03-master-01 kubelet[4604]: I0419 16:39:12.036736    4604 container_manager_linux.go:250] Creating Container Manager object based on Node Config: {RuntimeCgroupsName: SystemCgrApr 19 16:39:12 dotricorder03-master-01 kubelet[4604]: I041
Apr 19 16:39:12 dotricorder03-master-01 kubelet[4604]: I0419 16:39:12.036948    4604 kubelet.go:265] Watching apiserver
lines 1-60

**Comment by ncdc** [comment]
@csarora please attach the full output, either here or as a gist or to fpaste.org etc.

**Comment by csarora** [comment]
Attached file
[kubelet-logs.txt](https://github.com/kubernetes/kubernetes/files/936932/kubelet-logs.txt)

**Comment by ncdc** [comment]
The logs go from 16:38:15 to 17:39:12 UTC and they also appear to have some lines cut off. I don't see anything related to exec, but it's hard to tell because I'm not sure what time you ran exec, and the line cutoff issue. Could you please try this again and:

1) Make a note of the time you attempt `kubectl exec`. Feel free to try it a few times.
2) Get the logs from the apiserver and the kubelet during that time period. Please make sure lines are not cut off.

You can always do e.g. `sudo journalctl -u kubelet` > kubelet.log and then upload the full log by using the `gist` command line tool or any other means you have to get the file from your system to github or fpaste or wherever.

**Comment by csarora** [comment]
@ncdc https://gist.githubusercontent.com/csarora/e0edeea630fff8c8432723dc0ea3ac7f/raw/2dab7756c199fdb03ae694ab2a96dece3b417443/kubelet-log.txt

**Comment by ncdc** [comment]
@csarora your log file ends at 10:31 UTC, and you issued the exec 40 minutes after.

**Comment by csarora** [comment]
@ncdc...sorry i guess there is some confusion..time mentioned by me is incorrect....ran the command before...

**Comment by ncdc** [comment]
OK. I am swamped right now but can try to look at it next week if nobody
else gets back sooner.

**Comment by timstclair** [comment]
There should be a log line that looks like this:

```
Apr 20 18:03:02 gke-test-aa-default-pool-4ba465fc-1ff5 kubelet[1398]: I0420 18:03:02.632194    1398 server.go:740] POST /exec/default/busybox/busybox?command=sh&error=1&input=1&output=1&tty=1: (12.584911521s) hijacked [[kubectl/v1.5.3 (linux/amd64) kubernetes/029c3a4] 10.240.0.4:42948]
```

But in the log you posted there is not. This either indicates the the problem is in the API server and it's never contacting the Kubelet, or the exec call happened outside the timeframe of the log file.

**Comment by csarora** [comment]
@timstclair I feel its not reaching api server as i tried multiple times to run exec command but see no change in log

**Comment by lnquy** [comment]
@csarora: I got exactly the same error when running `minikube` k8s under a proxy.
I tried to use `curl` to connect to the `kubectl` API server and got `curl: (56) Received HTTP code 504 from proxy after CONNECT`. 
The problem is I already set the `env no_proxy` for the minikube network which can be access normally by using browsers but some how the request from `curl` still went to the proxy first and failed (of course).
So I think the problem may not from k8s itself but the underlying `curl` proxy.
Still don't figure out how to work around / solve this problem yet :tired_face:
